app/assets/javascripts/pretty-text/engines/discourse-markdown/image-controls.js

```diff
@@ -65,9 +65,49 @@ function buildScaleButton(selectedScale, scale) {
   );
 }
 
+function buildImageAltTextButton(altText) {
+  return `
+<span class="alt-text-readonly-container">
+  <span class="alt-text" aria-label="alt text">${altText}</span>
+  <span class="alt-text-edit-btn"><svg aria-hidden="true" class="fa d-icon d-icon-pencil svg-icon svg-string"><use xlink:href="#pencil-alt"></use></svg></span>
+</span>
+<input class="alt-text-input" hidden="true" type="text" value="${altText}" />
```

Comment:
I had the option to 
- put all elements here and hide/show them accordingly (which is what I'm doing here), 
- or dynamically add/remove elements in `composer-editor`

I chose the former (this) as it seemed to be lower in complexity.

Reply:
Also, I looked around for how `aria-labels` are applied but don't seem to find a lot of them. I've also hard coded the value but it seems like theres a couple of occurrences of translated aria labels. Would be nice to get some input here.

Reply:
I'm not sure I understand your question, but you could make it dynamic using I18n.t and concerning the content of the key I think I would go with something descriptive: "alt text used for the above image" , what do you think?

Reply:
> I'm not sure I understand your question

Haha you totally did.

That would work. I'd be curious how the whole i18n process is like in discourse, since I'm guessing there'd be some turnaround time for the rest of the translations to be ready.